tex/generic/pgf/utilities/pgfcalendar.code.tex

```diff
@@ -690,13 +695,13 @@
 
 \def\pgfcalendarshorthand#1#2{\csname pgfcalendar@shorthand@#1#2\endcsname}
 \expandafter\def\csname pgfcalendar@shorthand@d-\endcsname{%
-  {\pgfutil@tempcnta=\pgfcalendarcurrentday\relax\the\pgfutil@tempcnta}}
+  \if0\pgfcalendarcurrentday\else\pgfcalendarcurrentday\fi}
```

Comment:
This looks dangerous. It could fail in all kinds of situations, especially if `\pgfcalendarcurrentday` is accidentally empty.
```suggestion
  \ifnum\pgfcalendarcurrentday=0\relax\else\expandafter\pgfcalendarcurrentday\fi}
```

Reply:
The previous code already fails when `\pgfcalendarcurrentday` is empty:
```latex
\pgfutil@tempcnta=\relax
```
will be `Missing number, treated as zero`.

But why should it be empty? The manual states that `\pgfcalendarshorthand` "can be used inside a `\pgfcalendar`" and `\pgfcalendar` sets up all `\pgfcalendarcurrent<x>` macros.

The `\if0` will either strip a preceding `0` or just expand to `\pgfcalendarcurrentday`. Again, or just `\the\numexpr\pgfcalendarcurrentday\relax` but this will also fail when `\pgfcalendarcurrentday` is empty or undefined, of course.

Your change doesn't strip a preceding `0` (and `\pgfcalendarcurrentday` should never be `0` inside `\pgfcalendar`).